trieste/acquisition/function/function.py

```diff
@@ -42,6 +42,89 @@
 )
 
 
+class ProbabilityOfImprovement(SingleModelAcquisitionBuilder[ProbabilisticModel]):
+    """
+    Builder for the probability of improvement function, where the "best" value
+     is taken to be the minimum of the posterior mean at observed points.
+    """
+
+    def __repr__(self) -> str:
+        """"""
+        return "ProbabilityOfImprovement()"
+
+    def prepare_acquisition_function(
+        self, model: ProbabilisticModel, dataset: Optional[Dataset] = None
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: The data from the observer. Must be populated.
+        :return: The probability of improvement function. This function will raise
+            :exc:`ValueError` or :exc:`~tf.errors.InvalidArgumentError` if used with a batch size
+            greater than one.
+        :raise tf.errors.InvalidArgumentError: If ``dataset`` is empty.
+        """
+        tf.debugging.Assert(dataset is not None, [])
+        dataset = cast(Dataset, dataset)
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        mean, _ = model.predict(dataset.query_points)
+        eta = tf.reduce_min(mean, axis=0)
+        return probability_of_improvement(model, eta)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: The data from the observer.  Must be populated.
+        """
+        tf.debugging.Assert(dataset is not None, [])
+        dataset = cast(Dataset, dataset)
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        mean, _ = model.predict(dataset.query_points)
+        eta = tf.reduce_min(mean, axis=0)
+        function.update(eta)  # type: ignore
+        return function
+
+
+class probability_of_improvement(AcquisitionFunctionClass):
```

Comment:
do we need this? it is basically equivalent to probability_of_feasibility
 https://github.com/secondmind-labs/trieste/blob/develop/trieste/acquisition/function/function.py#L390

if I'm not mistaken?